### ISSUE TITLE
add __ prefix to local functions in (SPARC) assembly files

### ISSUE THREAD

**Issue by vladak** [openssl/openssl#9229]
`crypto/ec/asm/ecp_nistz256-sparcv9.pl` has a bunch of functions that are local however do not follow the common naming scheme, namely the `__` prefix. This needs to be addressed across all (SPARC) assembly files.

For starters: [sparc_asm_locl.patch.gz](https://github.com/openssl/openssl/files/3319919/sparc_asm_locl.patch.gz)

This is a follow-up from PR #9132.

**Comment by levitte** [comment]
"needs" is a relative term. It's a convention, not something that will make the compiler emit different code (apart from the symbol name), so this change is cosmetic. 

Something to keep in mind is that initial underscore is reserved in C, so this might be controversial to some.

A question is if this should be done only for solaris / sparc, or if other assembler sources should be similarly touched.

**Comment by levitte** [comment]
Note that I have nothing against such an effort.

**Comment by vladak** [comment]
> A question is if this should be done only for solaris / sparc, or if other assembler sources should be similarly touched.

Yep, that's why it's in the brackets - until the scope determined.

**Comment by vladak** [comment]
> "needs" is a relative term. It's a convention, not something that will make the compiler emit different code (apart from the symbol name), so this change is cosmetic.

I agree it's cosmetics. Arguably, this might have prevented #8936.

> Something to keep in mind is that initial underscore is reserved in C, so this might be controversial to some.

Double or single underscore, it should be consistent.

**Comment by nhorman** [comment]
Marking as inactive, to be closed at the end of 3.4 dev barring further input

**Comment by t8m** [comment]
Closing as the issue was labeled as inactive for extended period of time and there was no additional input.
Please add a comment if you have any further information to support reopening the issue.
